pyrokinetics/gk_code/GKOutputReaderCGYRO.py

```diff
@@ -260,59 +290,69 @@ def _set_fields(
 
             # If nonlinear, we can simply save the fields and continue
             if gk_input.is_nonlinear():
-                fields[ifield, :, :, :, :] = field_data.swapaxes(0, 1)
-                continue
-
-            # If theta_plot != theta_grid, we get eigenfunction data and multiply by the
-            # field amplitude
-            if data.ntheta_plot != data.ntheta_grid:
-                # Get eigenfunction data
-                raw_eig_data = raw_data.get(f"eigenfunctions_{field_name}", None)
-                if raw_eig_data is None:
-                    logging.warning(
-                        f"When setting fields, eigenfunction data for {field_name} not "
-                        f"found, expected the file bin.cygro.{field_name}b to exist. "
-                        f"Setting the field {field_name} to 0."
+                fields = field_data.swapaxes(0, 1)
+            else:
+                # If theta_plot != theta_grid, we get eigenfunction data and multiply by the
```

Comment:
Something to consider: is this block large/complex enough to pull out into a separate helper function?